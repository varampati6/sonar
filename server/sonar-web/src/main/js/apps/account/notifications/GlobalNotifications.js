/*
 * SonarQube
 * Copyright (C) 2009-2017 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import React from 'react';
import { connect } from 'react-redux';
import NotificationsList from './NotificationsList';
import { translate } from '../../../helpers/l10n';
import {
  getGlobalNotifications,
  getNotificationChannels,
  getNotificationGlobalTypes
} from '../../../store/rootReducer';
import type {
  Notification,
  NotificationsState,
  ChannelsState,
  TypesState
} from '../../../store/notifications/duck';
import { addNotification, removeNotification } from './actions';

type Props = {
  notifications: NotificationsState,
  channels: ChannelsState,
  types: TypesState,
  addNotification: (n: Notification) => void,
  removeNotification: (n: Notification) => void
};

function GlobalNotifications(props: Props) {
  return (
    <section>
      <h2 className="spacer-bottom">
        {translate('my_profile.overall_notifications.title')}
      </h2>

      <table className="form">
        <thead>
          <tr>
            <th />
            {props.channels.map(channel => (
              <th key={channel} className="text-center">
                <h4>{translate('notification.channel', channel)}</h4>
              </th>
            ))}
          </tr>
        </thead>

        <NotificationsList
          notifications={props.notifications}
          channels={props.channels}
          types={props.types}
          checkboxId={(d, c) => `global-notification-${d}-${c}`}
          onAdd={props.addNotification}
          onRemove={props.removeNotification}
        />
      </table>
    </section>
  );
}

const mapStateToProps = state => ({
  notifications: getGlobalNotifications(state),
  channels: getNotificationChannels(state),
  types: getNotificationGlobalTypes(state)
});

const mapDispatchToProps = { addNotification, removeNotification };

export default connect(mapStateToProps, mapDispatchToProps)(GlobalNotifications);

export const UnconnectedGlobalNotifications = GlobalNotifications;
