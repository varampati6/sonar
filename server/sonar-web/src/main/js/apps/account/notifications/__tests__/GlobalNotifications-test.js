/*
 * SonarQube
 * Copyright (C) 2009-2017 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import React from 'react';
import { shallow } from 'enzyme';
import { UnconnectedGlobalNotifications } from '../GlobalNotifications';

it('should match snapshot', () => {
  const channels = ['channel1', 'channel2'];
  const types = ['type1', 'type2'];
  const notifications = [
    { channel: 'channel1', type: 'type1' },
    { channel: 'channel1', type: 'type2' },
    { channel: 'channel2', type: 'type2' }
  ];

  expect(
    shallow(
      <UnconnectedGlobalNotifications
        notifications={notifications}
        channels={channels}
        types={types}
        addNotification={jest.fn()}
        removeNotification={jest.fn()}
      />
    )
  ).toMatchSnapshot();
});
