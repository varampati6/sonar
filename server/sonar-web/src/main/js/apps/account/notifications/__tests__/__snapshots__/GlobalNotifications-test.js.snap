// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<section>
  <h2
    className="spacer-bottom"
  >
    my_profile.overall_notifications.title
  </h2>
  <table
    className="form"
  >
    <thead>
      <tr>
        <th />
        <th
          className="text-center"
        >
          <h4>
            notification.channel.channel1
          </h4>
        </th>
        <th
          className="text-center"
        >
          <h4>
            notification.channel.channel2
          </h4>
        </th>
      </tr>
    </thead>
    <NotificationsList
      channels={
        Array [
          "channel1",
          "channel2",
        ]
      }
      checkboxId={[Function]}
      notifications={
        Array [
          Object {
            "channel": "channel1",
            "type": "type1",
          },
          Object {
            "channel": "channel1",
            "type": "type2",
          },
          Object {
            "channel": "channel2",
            "type": "type2",
          },
        ]
      }
      onAdd={[Function]}
      onRemove={[Function]}
      types={
        Array [
          "type1",
          "type2",
        ]
      }
    />
  </table>
</section>
`;
